### DISCUSSION TITLE
Is it possible to generate `typed-router.d.ts` without launching dev server?

### DISCUSSION THREAD

**Discussion by kingyue737** [posva/unplugin-vue-router#253] [Ideas]
I want to generate type declarations in `postinstall` hook like `nuxt prepare`. Then types will be updated once I run `pnpm update` and I can safely exclude auto-generated files from repo.

**Comment by posva** [comment]
It’s currently not possible but I would like to have something like this. I think I asked about it in unplugin (we need to run the things through the vite pipeline). There is maybe something that can be copied from nuxt

**Comment by angelhdzmultimedia** [comment]
Hi, I have curiosity about this.

In nuxt we can use  `experimental: { typedPages: true }` and will have completions of the routes and params.

![image](https://github.com/posva/unplugin-vue-router/assets/10191085/577397ec-ee59-4bc5-883d-7dac83d2a4b0)


Or is this question related to pure Vue project?
